rosbag2_py/CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+cmake_minimum_required(VERSION 3.5)
+project(rosbag2_py)
+
+#add_definitions(-D_SRC_RESOURCES_DIR_PATH="${CMAKE_CURRENT_SOURCE_DIR}/resources")
+
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_ros REQUIRED)
+find_package(pybind11 REQUIRED)
+find_package(rosbag2_cpp REQUIRED)
+find_package(rosbag2_compression REQUIRED)
+
+ament_python_install_package(${PROJECT_NAME})
+
+pybind11_add_module(_rosbag2_py SHARED
+  src/rosbag2_py/_rosbag2_py.cpp
+)
+# TODO(jacobperron): Consider modifying ament_python_install_module to accept targets
+# ament_python_install_module(_rosbag2_py
+#   DESTINATION_SUFFIX "rosbag2_py"
+# )
+# target_link_libraries(_rosbag2_py
+
+ament_target_dependencies(_rosbag2_py PUBLIC
+  "rosbag2_cpp"
+  "rosbag2_compression"
+)
+
+# Install cython modules as sub-modules of the project
+install(
+  TARGETS _rosbag2_py
+  DESTINATION "${PYTHON_INSTALL_DIR}/${PROJECT_NAME}"
+)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+  find_package(ament_cmake_pytest REQUIRED)
+
+  set(_PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE}")
+  if(WIN32 AND "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+    set(_PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE_DEBUG}")
+  endif()
+  set(pythonpath "${CMAKE_CURRENT_BINARY_DIR}/rosbag2_py;${CMAKE_CURRENT_SOURCE_DIR}")
+  if(NOT WIN32)
+    string(REPLACE ";" ":" pythonpath "${pythonpath}")
+  endif()
```

Comment:
I didn't test on windows but is this necessary?
Just wanted to know since I haven't seen other packages require this.

Reply:
what's the status on this?

Reply:
I asked around a bit. It turns out for the first if-statement, I needed to add a dependency on `python_cmake_module` in order for `PYTHON_EXECUTABLE` variable to be defined. That is added in 133e1a4. I will run Windows CI for debug build type to check what happens.

The second if-statement is for `pythonpath` to be defined, which is necessary.

Reply:
Oh Windows CI is actually failing on cannot find pybind first. Duh. The cmake module for finding pybind needs to be done first before I can test this variable in Windows.

Reply:
@mabelzhang are you looking into this cmake module? I would expect the vendor package to provide this.

Reply:
It was using the `pybind11_vendor` package before, but I think we want to switch to `pybind11-dev`, see this thread https://github.com/ros2/rosbag2/pull/308#discussion_r441850368

Reply:
can this be changed/removed now that we have moved back to the vendor package?

Reply:
These are both required as in https://github.com/ros2/rosbag2/pull/308#discussion_r455455913
Windows CI looks fine with the change in 133e1a4, so I think this is good the way it is.